scriptworker/cot/verify.py

```diff
@@ -965,6 +1014,57 @@ def verify_firefox_decision_command(decision_link):
 async def verify_decision_task(chain, link):
     """Verify the decision task Link.
 
+    This should be called in conjunction with ``verify_parent_task``.
+
+    Currently this function just verifies the command; in the future we should
+    rebuild the decision task definition from source, via json-e.
+
+    Args:
+        chain (ChainOfTrust): the chain we're operating on.
+        link (LinkOfTrust): the task link we're checking.
+
+    Raises:
+        CoTError: on chain of trust verification error.
+
+    """
+    # TODO: rebuild from json-e.
+    verify_firefox_decision_command(link, mach_commands=DECISION_MACH_COMMANDS)
+
+
+# verify_action_task {{{1
+async def verify_action_task(chain, link):
+    """Verify the action task Link.
+
+    This should be called in conjunction with ``verify_parent_task``.
+
+    Currently this function just verifies the command; in the future we should
+    rebuild the action task definition from the decision task's ``actions.json``
+    and ``task.extra.action`` via json-e.
+
+    Args:
+        chain (ChainOfTrust): the chain we're operating on.
+        link (LinkOfTrust): the task link we're checking.
+
+    Raises:
+        CoTError: on chain of trust verification error.
+
+    """
+    # TODO: download the ``actions.json`` from the decision / parent task;
+    # rebuild from ``task.extra.action`` via json-e
+    verify_firefox_decision_command(link, mach_commands=ACTION_MACH_COMMANDS)
+
+
+# verify_parent_task {{{1
+async def verify_parent_task(chain, link):
+    """Verify the parent task Link.
+
+    Action task verification is currently in the same verification function as
+    decision tasks, because sometimes we'll have an action task masquerading as
+    a decision task, e.g. in templatized actions for release graphs. To make
+    sure our guess of decision or action task isn't fatal, call this function
```

Comment:
Outdated comment? `verify_parent_task` already calls `is_action()` and the right `verify_*_task()`. 

If not, I'm not sure to understand why calling these function once more adds more security.

Reply:
This is an explanation of what this does, rather than telling users to do additional steps. I can clarify.